uxarray/grid/integrate.py

```diff
@@ -1,184 +1,356 @@
 
-def _get_zonal_faces_weight_at_constLat(
-    faces_edges_cart_candidate,
-    latitude_cart,
-    face_latlon_bound_candidate,
-    is_latlonface=False,
-    is_face_GCA_list=None,
-):
-    """Utilize the sweep line algorithm to calculate the weight of each face at
-    a constant latitude.
 
-     Parameters
+def _is_edge_gca(is_GCA_list, is_latlonface, edges_z):
+    """Determine if each edge is a Great Circle Arc (GCA) or a constant
+    latitude line in a vectorized manner.
+
+    Parameters:
     ----------
-    faces_edges_cart_candidate : np.ndarray
-        A list of the candidate face polygon represented by edges in Cartesian coordinates.
-        The faces must be selected in the previous step such that they will be intersected by the constant latitude.
-        It should have the same shape as the face_latlon_bound_candidate.
-        The input should not contain any 'INT_FILL_VALUE'. Shape: (n_faces(candidate), n_edges, 2, 3)
+    is_GCA_list : np.ndarray or None
+        An array indicating whether each edge is a GCA (True) or a constant latitude line (False).
+        Shape: (n_edges). If None, edge types are determined based on `is_latlonface` and the z-coordinates.
+    is_latlonface : bool
+        Flag indicating if all edges should be considered as lat-lon faces, which implies all edges
+        are either constant latitude or longitude lines.
+    edges_z : np.ndarray
+        Array containing the z-coordinates for each vertex of the edges. This is used to determine
+        whether edges are on the equator or if they are aligned in latitude when `is_GCA_list` is None.
+        Shape should be (n_edges, 2).
 
-    latitude_cart : float
-        The latitude in Cartesian coordinates (The normalized z coordinate)
+    Returns:
+    -------
+    np.ndarray
+        A boolean array where each element indicates whether the corresponding edge is considered a GCA.
+        True for GCA, False for constant latitude line.
+    """
+    if is_GCA_list is not None:
+        return is_GCA_list
+    if is_latlonface:
+        return ~np.isclose(edges_z[:, 0], edges_z[:, 1], atol=ERROR_TOLERANCE)
+    return ~(
+        np.isclose(edges_z[:, 0], 0, atol=ERROR_TOLERANCE)
+        & np.isclose(edges_z[:, 1], 0, atol=ERROR_TOLERANCE)
+    )
 
-    face_latlon_bound_candidate : np.ndarray
-        The list of latitude and longitude bounds of candidate faces.
-        It should have the same shape as the face_edges_cart_candidate.
-        Shape: (n_faces(candidate),,2, 2),
-        [...,[lat_min, lat_max], [lon_min, lon_max],...]
 
-    is_latlonface : bool, optional, default=False
-        A global flag to indicate if faces are latlon face. If True, then treat all faces as latlon faces. Latlon face means
-        That all edge is either a longitude or constant latitude line. If False, then all edges are GCA.
-         Default is False.
+def get_non_conservative_zonal_face_weights_at_const_lat_original(
```

Comment:
A few comments on this:
- Why is this in the user API?
- The naming is way too long, especially for an user API. It feels like it should have been something like `get_zonal_face_weights` with arguments `method` (with options "conservative" and "non-conservative") and `cross-section` with options "constant_lat" and "constant_lon"
- It is pretty confusing to the user that there is an "original" function and there is one with a normal name (without "original") in the API
- If it will have to be kept in the user API, I am not sure about such a function returning `polars.dataframe` would be  a useful scenario

Reply:
> Why is this in the user API?

I will convert it back to internal; it wasn't the intention to expose it to the user.

> The naming is way too long, especially for an user API. It feels like it should have been something like get_zonal_face_weights with arguments method (with options "conservative" and "non-conservative") and cross-section with options "constant_lat" and "constant_lon"

This is fair, I can try to trim down the names of the function.

Reply:
As you will convert it back to internal, I am less concerned about the naming unless you have an immediate idea to implement